### PR TITLE
Fix bug related to acquisition batch variable

### PR DESCRIPTION
**Pull request by b5y**
#### Summary:
Fitting the surrogate model might crash if user gives wrong model parameters. Instead of crashing, current solution gives an error and says that model is incompatible.

---
This change is [Reviewable](https://reviewable.io/reviews/elfi-dev/elfi/293)

### CODE REVIEW

File: elfi/methods/bo/acquisition.py [hunk 1]
```diff
@@ -548,6 +550,10 @@ def _evaluate_logpdf(theta):
             batch_theta = samples[-n:, :]
             break
 
+        # check if the acquisition batch is still filled by zero values
+        if not np.any(batch_theta):
+            raise ValueError('Incompatible model. Please check the parameters of your model.')
```
Comment (hpesonen) [inline elfi/methods/bo/acquisition.py]: Isn't this actually already implemented in [L512](https://github.com/elfi-dev/elfi/blob/47a39e52c4a02e4040518f0f0216558859733b21/elfi/methods/bo/acquisition.py#L512)? There just seems to be a bug as the condition in L512 is never `True`? However, 'Unable to find a suitable initial point.' is a bit more informative than 'Incompatible model. Please check the parameters of your model.'

Comment (b5y) [reply elfi/methods/bo/acquisition.py]: No, it is not the same. If you use variable in the specific scope with some condition (like [`batch_theta`](https://github.com/elfi-dev/elfi/blob/dev/elfi/methods/bo/acquisition.py#L548) in `for` loop in our example) and try to call it outside of this scope (outside of `for` loop), then you **have to define** this variable before your scope.

I don't know how this code was accepted before me, but any `Python` developer **must** know [**this simple rules**](https://stackoverflow.com/questions/291978/short-description-of-the-scoping-rules).

Comment (hpesonen) [reply elfi/methods/bo/acquisition.py]: I'm not saying it's the same thing. I'm pointing out the bug in the condition in L512. Without the bug, one would proceed to `raise SystemExit("Unable to find a suitable initial point.")` if one doesn't find any non-faulty initial values and proceed to the samplers. I AM suggesting that you'd use the error msg "Unable to find a suitable initial point."  and also you could remove L512-L513 if you'd prefer predefining batch_theta as you have.

Comment (wleoncio) [reply elfi/methods/bo/acquisition.py]: Would a compomise resolve this impasse? Maybe have the errors (either or both, I can't tell) read something like `"Unable to find a suitable initial point or incompatible model. Please check the parameters"`.

**Pull request merged**